src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -382,4 +683,261 @@ def truncate_colormap(
+            # seconds by using the ground track velocity (meters/second).
+
+            # Use the aztime of each target in radar coordinates to
+            # interpolate the orbit in order to get the
+            # platform position and velocity in ECEF coordinates.
+            platform_pos, platform_vel = orbit.interpolate(aztime)
+
+            grd_trk_vel_at_arrow_tail = nisarqa.get_ground_track_velocity(
+                target_xyz,  # ECEF position of target on the ground
+                platform_pos,  # az
+                platform_vel,
+            )
+
+            az_off_val_at_arrow_tail_in_seconds = (
+                az_offsets_at_arrow_tails[i, j] / grd_trk_vel_at_arrow_tail
+            )
+
+            # Shift the radar grid's azimuth values
+            aztime += az_off_val_at_arrow_tail_in_seconds
+
+            # Shift the radar grid's range values
+            # Note: range offset values were already in meters, so just add them
+            srange += rg_offsets_at_arrow_tails[i, j]
+
+            # 4) Run rdr2geo to convert back to (x_1, y_1) LLH
+
+            # Skip NaN pixels (this is usually geocoding fill)
+            if not np.isfinite(aztime) or not np.isfinite(srange):
+                x_tail_coords_shifted[i, j] = np.nan
+                y_tail_coords_shifted[i, j] = np.nan
+                continue
+
+            xyz = isce3.geometry.rdr2geo_bracket(
+                # az time in seconds since orbit.reference_epoch
+                aztime=aztime,
+                # slant range in meters
+                slant_range=srange,
+                orbit=orbit,
+                side=look_side,
+                # Here, doppler is a scalar value -- not a LUT. Because
+                # we know the target azimuth/range coordinates so we don't
+                # need Doppler as a function of range & azimuth, it can
+                # just be a scalar value. Since we care about the Doppler
+                # of the image grid, use 0.0.
+                doppler=0.0,
+                # Technically the wavelength is not needed here since
+                # Doppler is zero and the wavelength is just multiplied
+                # by the Doppler. But since we already need the wavelength
+                # for geo2rdr, might as well use it.
+                wavelength=wavelength,
+                # Ideally we would use the actual DEM, but we don't have
+                # this information available so use a zero-height DEM.
+                dem=isce3.geometry.DEMInterpolator(),
+            )
+
+            # Once again, ignore the returned height
+            lon, lat, _ = ellipsoid.xyz_to_lon_lat(xyz)
+
+            # 5) Convert shifted LLH (x_1, y_1) back into projected
+            #    coordinates
+            # To convert from LLH back to the projected coordinates of the
+            # input arrays, use the forward() method of the same `proj`
+            # object we created above.
+            xyz = proj.forward([lon, lat, 0])
```

Comment:
Let's not call this `xyz`. We use the variable name `xyz` above to represent the location of the shifted target in ECEF coordinates. Now we're using it to represent the location in projected coordinates.

```suggestion
            x_shift, y_shift, _ = proj.forward([lon, lat, 0])
```